doc/source/reference/routines.rst

```diff
@@ -42,6 +42,7 @@ indentation.
    routines.padding
    routines.polynomials
    routines.random
```

Comment:
So this 'routines.random' line should be removed, right @mattip? Along with file https://github.com/mattip/numpy/blob/randomgen/doc/source/reference/routines.random.rst

Reply:
The file should not be removed. The link should be moved to `doc/source/reference/random/index.rst` and the difference between `numpy.random.X` and `numpy.random.gen.X` should be made clear. I would like to do a side-by-side comparison, but the page is automatically generated from the docstrings.

Reply:
@mattip I see that there is this file doc/source/reference/random/legacy.rst

Reply:
OK. Thanks, finally got it. I will remove it. We should find a better way to expose the legacy document. Right now it is [hidden](https://6747-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/random/index.html#random-generator)